lib/google_search_data_viewer_web/helpers/keyword_helper.ex

```diff
@@ -0,0 +1,34 @@
+defmodule GoogleSearchDataViewerWeb.KeywordHelper do
+  alias NimbleCSV.RFC4180, as: CSV
+
+  @max_keyword_upload_count 1000
+
+  def validate_and_parse_keyword_file(file) do
+    with true <- file_valid?(file),
+         {:ok, keywords} <- parse_keyword_file(file) do
+      {:ok, keywords}
+    else
+      false -> {:error, :invalid_extension}
+      :error -> {:error, :invalid_length}
+    end
+  end
+
+  defp file_valid?(file), do: file.content_type == "text/csv"
+
+  defp parse_keyword_file(file) do
+    keywords =
+      file.path
+      |> File.stream!()
+      |> CSV.parse_stream(skip_headers: false)
+      |> Enum.to_list()
+      |> List.flatten()
+
+    keywords_length = length(keywords)
```

Comment:
```suggestion
    keywords_length = Enum.count(keywords)
```

We refer to use `Enum.count/1` instead of `length/1`